profiling/src/profile/mod.rs

```diff
@@ -463,7 +469,13 @@ impl Profile {
 
         let mut buffer: Vec<u8> = Vec::new();
         profile.encode(&mut buffer)?;
-        Ok(EncodedProfile { start, end, buffer })
+
+        Ok(EncodedProfile {
+            start,
+            end,
+            buffer,
+            endpoints_stats: self.endpoints.stats.clone().into(),
+        })
```

Comment:
>Is there any limitation into exposing the actual ProfiledEndpointsStats opaque map-like object over the FFI?

Reply:
When I try:
```
#[repr(C)]
pub struct EncodedProfile {
    start: Timespec,
    end: Timespec,
    buffer: ddcommon_ffi::Vec<u8>,
    endpoints_stats: profiles::profiled_endpoints::ProfiledEndpointsStats,
}
```
I get this: 
![image](https://user-images.githubusercontent.com/17292193/201888104-5929df6e-6afb-46d0-b3c8-eb69b63e0648.png)

Reply:
My rust-foo sucks as usual but I think it's because the representation of the `IndexMap` is not ffi-able and we're returning it "by value". I think the trick is returning a pointer to it instead, something along the lines of (incomplete example):

```diff
diff --git a/profiling-ffi/src/profiles.rs b/profiling-ffi/src/profiles.rs
index b160731..64a002d 100644
--- a/profiling-ffi/src/profiles.rs
+++ b/profiling-ffi/src/profiles.rs
@@ -4,6 +4,7 @@
 use crate::Timespec;
 use datadog_profiling::profile as profiles;
 use ddcommon::profiled_endpoints::ProfiledEndpointStats;
+use ddcommon::profiled_endpoints::ProfiledEndpointsStats;
 use ddcommon_ffi::slice::{AsBytes, CharSlice, Slice};
 use std::convert::{TryFrom, TryInto};
 use std::str::Utf8Error;
@@ -375,6 +376,7 @@ pub struct EncodedProfile {
     end: Timespec,
     buffer: ddcommon_ffi::Vec<u8>,
     endpoints_stats: ddcommon_ffi::Vec<ProfiledEndpointStats>,
+    endpoints_stats_v2: *const ddcommon::profiled_endpoints::ProfiledEndpointsStats,
 }

 impl From<datadog_profiling::profile::EncodedProfile> for EncodedProfile {
@@ -383,12 +385,14 @@ impl From<datadog_profiling::profile::EncodedProfile> for EncodedProfile {
         let end = value.end.into();
         let buffer = value.buffer.into();
         let endpoints_stats = value.endpoints_stats.into();
+        let endpoints_stats_v2 = Box::into_raw(Box::new(value.endpoints_stats_v2));

         Self {
             start,
             end,
             buffer,
             endpoints_stats,
+            endpoints_stats_v2,
         }
     }
 }
diff --git a/profiling/src/profile/mod.rs b/profiling/src/profile/mod.rs
index b7ab9e9..f8d242f 100644
--- a/profiling/src/profile/mod.rs
+++ b/profiling/src/profile/mod.rs
@@ -211,6 +211,7 @@ pub struct EncodedProfile {
     pub end: SystemTime,
     pub buffer: Vec<u8>,
     pub endpoints_stats: Vec<ProfiledEndpointStats>,
+    pub endpoints_stats_v2: ProfiledEndpointsStats,
 }

 impl Endpoints {
@@ -475,6 +476,7 @@ impl Profile {
             end,
             buffer,
             endpoints_stats: self.endpoints.stats.clone().into(),
+            endpoints_stats_v2: self.endpoints.stats.clone(),
         })
     }
```

Ping me if you get stuck, we can get stuck together for more fun! ;)

Reply:
Nice I'll give it a try. Thx